lib/galaxy/tools/verify/__init__.py

```diff
@@ -263,9 +273,21 @@ def get_lines_diff(diff):
 
 def files_re_match(file1, file2, attributes=None):
     """Check the contents of 2 files for differences using re.match."""
-    local_file = open(file1, 'rb').readlines()  # regex file
-    history_data = open(file2, 'rb').readlines()
-    assert len(local_file) == len(history_data), 'Data File and Regular Expression File contain a different number of lines (%d != %d)\nHistory Data (first 40 lines):\n%s' % (len(local_file), len(history_data), ''.join(unicodify(history_data[:40])))
+    join_char = ''
+    to_strip = os.linesep
```

Comment:
This isn't correct, we want to always strip carriage returns here

Reply:
Do we care about Windows where ``os.linesep`` would be ``\r\n`` (CR LF)? Probably not.

However, more realistically, we probably should care about files using Windows new lines - the Python default universal text mode ought to take care of that case for us (although we'd have to double check Python 2 vs 3 there), that might be a good reason not to use binary mode for the line-based ``re`` check?

Reply:
We actually want to always strip `\r\n` here, independent of what platform we're on. If old test data ends with `\r\n` and new data produced by a tool run is `\n` we don't want to fail IMO. We haven't in the past, and I don't think we should introduce this now.

Reply:
Actually `\r\n` is being stripped by readlines, it'll just end up as `\n`, so that's actually the correct change. ...